src/FakeItEasy/MethodInfoExtensions.cs

```diff
@@ -10,6 +11,8 @@ namespace FakeItEasy
     /// </summary>
     internal static class MethodInfoExtensions
     {
+        private static readonly ConcurrentDictionary<MethodInfo, ObjectMethod> ObjectMethodMap = new ConcurrentDictionary<MethodInfo, ObjectMethod>();

```

Comment:
I think this is a good use case for [target-typed new expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new)